python/beeai_framework/context.py

```diff
@@ -156,8 +156,10 @@ async def handler() -> R:
                 await emitter.emit("start", None)
 
                 async def _context_storage_run() -> R:
-                    storage.set(context)
-                    return await fn(context)
+                    storage_token = storage.set(context)
+                    response = await fn(context)
+                    storage.reset(storage_token)
+                    return response
```

Comment:
IIUC before we were setting the store every time there was a run and overwriting the previous value, whereas now it resets the store back to its previous value after a run. I'm unsure if I'm misunderstanding the purpose of the store, but wasn't it a feature to be able to check the context of the most recent run? (which this would remove) or was that an unintended side effect we didn't mean to have?

Reply:
Good it. It is okay how it is.